pages/doing_api_first_development.md

```diff
@@ -11,96 +11,332 @@ sitemap:
+    @Override
+    public ResponseEntity<Pet> findPetById(Long id) {
+        // previously shown above
+    }
 
-When using Eclipse or VSCode
+    /**
+     * Implements the method used at the RequestMapping of /pets in the PetsApi.
+     * <p> 
+     * Returns an HTTP Status of 501 Not Implemented with an example response body defined 
+     * OpenAPI Specification file and generated by the OpenAPI Generator for Spring.
+     */
+    @Override
+    public ResponseEntity<List<Pet>> findPets(List<String> tags, Integer limit) {
+        return PetsApiDelegate.super.findPets(tags, limit);
+    }
 
-* With maven
-```bash
-./mvnw eclipse:clean eclipse:eclipse
+    private List<Pet> getPets() {
+        // previously shown above
+    }
+}
 ```
-When using IntelliJ
-* With maven
-```bash
-./mvnw idea:idea
+
+Restart the server and observe the following in a terminal (or "command prompt") with `curl` or other HTTP tooling, e.g., Insomina, Postman, httpie:
+
+```shell
+$ curl -H "Accept: application/json" http://localhost:8081/api/pets/1
+ {
+  "name" : "Chessie Cat",
+  "tag" : "cat",
+  "id" : 1
+}
+$ curl -H "Accept: application/json" http://localhost:8081/api/pets?tags=cat
+*   Trying [::1]:8081...
+* Connected to localhost (::1) port 8081
+> GET /api/pets?tags=cat HTTP/1.1
+> Host: localhost:8081
+> User-Agent: curl/8.4.0
+> Accept: application/json
+> 
+< HTTP/1.1 501 Not Implemented
+# additional headers trimmed for brevity
+< Content-Type: application/json; charset=UTF-8
+< Content-Length: 108
+< 
+{ [108 bytes data]
+* Connection #0 to host localhost left intact
+[ { "name" : "Chessie Cat", "id" : 1, "tag" : "cat" }, { "name" : "Chessie Cat", "id" : 1, "tag" : "cat" } ]
 ```
 
-### Using the `openapi-client` Sub-Generator
+The example body is returned though the HTTP Status of the response is `501 Not Implemented`.
+
+### Authentication
 
-JHipster also provides support for generation of client code using [Spring Cloud OpenFeign](https://docs.spring.io/spring-cloud-openfeign/docs/current/reference/html/) or Spring Webclient for reactive apps using an OpenAPI/Swagger specification.
-The generated Client can be used in both Monolithic and Micro-service applications and supports Swagger v2 and OpenAPI v3 definitions. To invoke this sub-generator run `jhipster openapi-client`.
+JHipster, by default, secures `/api/**` in `demo.jhipster.config.SecurityConfiguration#filterChain(HttpSecurity,  MvcRequestMatcher.Builder)` requiring authentication, e.g., `.requestMatchers(mvc.pattern("/api/**")).authenticated()`.
 
+Authentication is beyond the scope of this guide. However, review of the `demo.jhipster.security.jwt.TokenAuthenticationIT` integration test can provide some insight into testing authentication in Insomina, Postman, curl, and other HTTP tools. For example, it is possible to generate a JWT Bearer token using Docker as follows:
 
+```shell
+docker run --rm --name jwt-cli bitnami/jwt-cli encode \
+-S b64:<JHIPSTER_JWT_SECRET> \
```

Comment:
Maybe add a hint where to find the configured secret (or am I mistaken it must be a valid secret?)